src/runtime_src/core/common/drv/xrt_cu.c

```diff
@@ -259,6 +259,100 @@ static inline void process_pq(struct xrt_cu *xcu)
 		xcu->max_running = xcu->num_rq;
 }
 
+static inline void
+try_abort_cmd(struct xrt_cu *xcu, struct kds_command *abort_cmd)
+{
+	struct kds_command *xcmd;
+	struct kds_command *tmp;
+	u32 handle;
+
+	/* Never call this function on the performance critical path */
+	handle = *(u32 *)abort_cmd->info;
+	list_for_each_entry_safe(xcmd, tmp, &xcu->rq, list) {
+		if (xcmd->exec_bo_handle != handle)
+			continue;
+
+		/* Found the xcmd to abort! */
+		abort_cmd->status = KDS_COMPLETED;
+
+		xcu_info(xcu, "Abort command(%d) on running queue", handle);
+		xcmd->status = KDS_ABORT;
+		move_to_queue(xcmd, &xcu->cq, &xcu->num_cq);
+		--xcu->num_rq;
+		return;
+	}
+
+	list_for_each_entry_safe(xcmd, tmp, &xcu->sq, list) {
+		if (xcmd->exec_bo_handle != handle)
+			continue;
+
+		xcu_info(xcu, "Abort command(%d) on submitted queue", handle);
+		/* Found the xcmd to abort! */
+		if (!xcu->info.sw_reset) {
+			xcu_warn(xcu, "No sw resset. Device goto bad state");
+			abort_cmd->status = KDS_TIMEOUT;
```

Comment:
do we want to wait for a few seconds before claim that the cmd has timed out?

Reply:
I assume user would only send abort command down when he thought CU is hang or something is wrong. This is why I don't delay in here. I could wait a few seconds at this place.

Reply:
Oh, then you need to return KDS_ABORTED, not KDS_TIMEOUT.